draft-ietf-quic-load-balancers.md

```diff
@@ -324,6 +308,78 @@ the packet. The number of servers is usually consistent on the time scale of a
 QUIC connection handshake. Another might simply hash the address/port 4-tuple.
 See also {{version-invariance}}.
 
+Note that any information used by the Fallback Algorithm might not be repeated
+from one QUIC packet to another. The 4-tuple is assumed to remain stable during
+the handshake, but the server can change the destination connection ID after the
+client's first flight.
+
+Therefore, if the fallback algorithm uses information other than the 4-tuple in
+its routing decision, the load balancer SHOULD store the routing decision for
+that 4-tuple in order to route future packets unless doing so would exceed
+resource limits. It can free these records when the 4-tuple uses a routable CID.
+This state allows the load balancer to successfully route other packets from
+the client's first flight, as well as subsequent flights if the server provides
+an unroutable CID for the reasons described in {{config-failover}}.
+
+Even if the fallback algorithm solely uses the 4-tuple, it MAY store the
+algorithm result in case changes to the server pool change fallback algorithm
+output.
+
+The load balancer MAY also store the routing decision indexed by DCID. This
+is not robust to new server-selected DCIDs, but can increase robustness when
+a connection to a server generating unroutable CIDs experiences a NAT rebinding.
+An entry can be freed when the corresponding 4-tuple uses a different CID,
+whether routable or not.
+
```

Comment:
We seem to be dancing around the issue. First, NAT rebinding *during the client's first fly" ought to be fairly rare -- rare enough to be treated as an error case. Second, resilience to NAT rebinding requires routing based on DCID, instead of the 4 tuple. Why not just say that if a server uses unroutable CIDs, then NAT rebinding is not supported? And maybe tell servers that use unroutable CIDs to set the TP "disable_active_migration=TRUE" ?

Reply:
Whoops, the language about disable_active_migration and NEW_CONNECTION_ID fell out of the edit. I've restored it.